dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/program/notification/ProgramNotificationServiceTest.java

```diff
@@ -367,4 +376,67 @@ void eventDataValuesSearchStringTest() {
         events.stream().flatMap(e -> e.getEventDataValues().stream()).toList().contains(edv3),
         "should not contain this event data value");
   }
+
+  @Test
+  void testGetEnrollmentsWithScheduledNotifications() {
+    ProgramNotificationTemplate
+        a1 =
+            createProgramNotificationTemplate(
+                "a1",
+                -1,
+                SCHEDULED_DAYS_INCIDENT_DATE,
+                ProgramNotificationRecipient.TRACKED_ENTITY_INSTANCE),
+        a2 =
+            createProgramNotificationTemplate(
+                "a2",
+                1,
+                SCHEDULED_DAYS_INCIDENT_DATE,
+                ProgramNotificationRecipient.TRACKED_ENTITY_INSTANCE),
+        a3 =
+            createProgramNotificationTemplate(
+                "a3",
+                7,
+                SCHEDULED_DAYS_ENROLLMENT_DATE,
+                ProgramNotificationRecipient.TRACKED_ENTITY_INSTANCE);
+    programNotificationStore.save(a1);
+    programNotificationStore.save(a2);
+    programNotificationStore.save(a3);
+    // TE
+    TrackedEntity trackedEntityX = createTrackedEntity(organisationUnitA);
+    TrackedEntity trackedEntityY = createTrackedEntity(organisationUnitA);
+    trackedEntityService.addTrackedEntity(trackedEntityX);
+    trackedEntityService.addTrackedEntity(trackedEntityY);
+    // Program
+    programA.setNotificationTemplates(Sets.newHashSet(a1, a2, a3));
+    programService.updateProgram(programA);
+    // Dates
+    Calendar cal = Calendar.getInstance();
+    PeriodType.clearTimeOfDay(cal);
+    Date today = cal.getTime();
+    cal.add(Calendar.DATE, 1);
+    Date tomorrow = cal.getTime();
+    cal.add(Calendar.DATE, -2);
+    Date yesterday = cal.getTime();
+    cal.add(Calendar.DATE, -6);
+    Date aWeekAgo = cal.getTime();
+    // Enrollments
+    Enrollment enrollmentA = new Enrollment(today, tomorrow, trackedEntityX, programA);
+    enrollmentStore.save(enrollmentA);
```

Comment:
you should not introduce dependencies to our stores as we want to make them private. can you not save using the manager?